### ISSUE TITLE
Error: "An S3 ACL has not be initialized" when pushing changes

### ISSUE THREAD

**Issue by angeloashmore** [prismicio/slice-machine#958]
### Versions

- slice-machine-ui: `1.0.2`
- node: [unknown]

### Reproduction

N/A

### Steps to reproduce

Push changes to Prismic containing a new screenshot.

### What is expected?

Changes are pushed to Prismic.

### What is actually happening?

An error occurs when uploading the screenshots. See the following screenshot of the error.

![](https://user-images.githubusercontent.com/38848268/235800561-16af62fc-9c69-4ede-aa14-fffc50db85b2.png)

---

This error was originally reported by @draykefriesen and @tjkohli here: https://github.com/prismicio/slice-machine/issues/939#issuecomment-1532242808.

It's possible the error reported in #939 is related to the error reported here.

**Comment by angeloashmore** [comment]
I suggest changing `ScreenshotsManager.prototype.uploadScreenshot()` to initialize an S3 ACL rather than asserting that it was initialized. `initS3ACL()` is idempotent, so we can initialize it early _and_ just in time in case it wasn't already initialized.

https://github.com/prismicio/slice-machine/blob/25991210dccfe334f0a41ef31824e552f1025840/packages/manager/src/managers/screenshots/ScreenshotsManager.ts#L256

That that line can become:

```typescript
async uploadScreenshot(
	args: ScreenshotsManagerUploadScreenshotArgs,
): Promise<ScreenshotsManagerUploadScreenshotReturnType> {
	await this.initS3ACL();

	// ...
}
```

**Comment by hypervillain** [comment]
This should be fixed with version `1.0.3`. Thanks for your report!
